File: fix.dart
import 'dart:io';

void main() {
  final libDir = Directory(r'c:\Users\Acer\Desktop\Led Truck\ledtruck\lib');
  
  for (final file in libDir.listSync(recursive: true)) {
    if (file is File && file.path.endsWith('.dart')) {
      String content = file.readAsStringSync();
      String newContent = content;
      
      final regex = RegExp(r'\.withOpacity\(([^)]+)\)');
      newContent = newContent.replaceAllMapped(regex, (match) {
        return '.withValues(alpha: ${match.group(1)})';
      });
      
      final fileName = file.uri.pathSegments.last;
      
      if (fileName == 'modal_editar_dispositivo.dart' || fileName == 'modal_vincular_carro.dart') {
        final valReg = RegExp(r'(\s+)value: (selected[A-Za-z0-9_]+),');
        newContent = newContent.replaceAllMapped(valReg, (match) {
          return '${match.group(1)}initialValue: ${match.group(2)},';
        });
      }
      
      if (fileName == 'franqueado_campanhas_screen.dart') {
        newContent = newContent.replaceAll(RegExp(r"import 'package:go_router/go_router\.dart';\r?\n?"), "");
      }
      
      if (fileName == 'base_components.dart') {
        newContent = newContent.replaceAll('dialogBackgroundColor', 'dialogTheme.backgroundColor');
      }
      
      if (content != newContent) {
        file.writeAsStringSync(newContent);
        print('Fixed: ${file.path}');
      }
    }
  }
}

File: fix.py
import os
import re

lib_dir = r"c:\Users\Acer\Desktop\Led Truck\ledtruck\lib"

def run():
    for root, dirs, files in os.walk(lib_dir):
        for file in files:
            if file.endswith('.dart'):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()
                
                new_content = content
                
                # 2. .withOpacity -> .withValues(alpha: )
                new_content = re.sub(r'\.withOpacity\(([^)]+)\)', r'.withValues(alpha: \1)', new_content)
                
                # 3. value: -> initialValue:
                if file in ['modal_editar_dispositivo.dart', 'modal_vincular_carro.dart']:
                    new_content = re.sub(r'(\s+)value: (selected[A-Za-z0-9_]+),', r'\1initialValue: \2,', new_content)
                
                # 4. Remove go_router in franqueado_campanhas
                if file == 'franqueado_campanhas_screen.dart':
                    new_content = re.sub(r"import 'package:go_router/go_router\.dart';\r?\n?", "", new_content)
                
                # 5. dialogBackgroundColor -> dialogTheme.backgroundColor
                if file == 'base_components.dart':
                    new_content = new_content.replace('dialogBackgroundColor', 'dialogTheme.backgroundColor')
                
                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated: {filepath}")

if __name__ == "__main__":
    run()
